lib/core/src/Cardano/Wallet/Api/Types/SchemaMetadata.hs

```diff
@@ -0,0 +1,93 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE KindSignatures #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE NoMonomorphismRestriction #-}
+{-# LANGUAGE StrictData #-}
+
+-- |
+-- Copyright: © 2018-2022 IOHK
+-- License: Apache-2.0
+--
+-- A wrapper around TxMetadata to allow different JSON codecs. (ADP-1596)
+-- see https://github.com/input-output-hk/cardano-node/blob/master/cardano-api/src/Cardano/Api/TxMetadata.hs
+module Cardano.Wallet.Api.Types.SchemaMetadata where
+
+import Cardano.Api
+    ( Error (displayError)
+    , TxMetadataJsonSchema (..)
+    , metadataFromJson
+    , metadataToJson
+    )
+import Cardano.Wallet.Primitive.Types.Tx
+    ( TxMetadata )
+import Control.Applicative
+    ( liftA2, (<|>) )
+import Control.DeepSeq
+    ( NFData )
+import Data.Aeson
+    ( FromJSON (parseJSON), ToJSON (toJSON) )
+import GHC.Generics
+    ( Generic )
+import Prelude
+
+-- | a tag to select the json codec
+data TxMetadataSchema = TxMetadataNoSchema | TxMetadataDetailedSchema
+    deriving (Show, Eq, Generic, NFData)
+
+-- | a wrapper to drive the json codec of metadata
```

Comment:
(Tip for future: Treating haddock comments as full sentences looks better in the rendered documentation. Hence, starting with an upper case is preferred.)

Reply:
fixed